apps/user_ldap/lib/Connection.php

```diff
@@ -424,8 +425,14 @@ private function doCriticalValidation() {
 			(string)$this->configPrefix .'): ';
 
 		//options that shall not be empty
-		$options = ['ldapHost', 'ldapPort', 'ldapUserDisplayName',
+		$options = ['ldapHost', 'ldapUserDisplayName',
 			'ldapGroupDisplayName', 'ldapLoginFilter'];
+
+		//ldapPort should not be empty either unless ldapHost is pointing to a socket
+		if(substr($this->configuration->ldapHost, 0, strlen('ldapi://')) != 'ldapi://') {
```

Comment:
This test is done in a lot of places, could you add a method on configuration that contains this test?
```suggestion
		if ($this->configuration->usesLdapi()) {
```

Reply:
Done: https://github.com/nextcloud/server/pull/24574/commits/343ff3f37aed6f9ae147ea3d45c9843e23a84fd9